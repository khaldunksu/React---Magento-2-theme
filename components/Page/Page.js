import React from "react";
import {connect} from "react-redux";
import PropTypes from "prop-types";
import ErrorPage from "next/error";

export function PageComponent(props) {
    const {children, cmsError} = props;

    if (cmsError) {
        return <ErrorPage statusCode={404} />;
    }

    return (
        <div>
            <div className="container js-hook__main-layout">{children}</div>
        </div>
    );

}

PageComponent.propTypes = {
    children: PropTypes.oneOfType([PropTypes.array, PropTypes.object]),
};
PageComponent.defaultProps = {
    children: null,
};

export default connect(state => state)(PageComponent);
