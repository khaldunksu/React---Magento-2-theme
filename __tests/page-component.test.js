import { shallow } from "enzyme";
import React from "react";
import Router from "next/router";
import toJson from "enzyme-to-json";
import { PageComponent } from "../components/Page/Page";

const mockedRouter = { push: () => {} };
Router.router = mockedRouter;

describe("Component Page matches snapshot", () => {
    it("renders page layout correctly", () => {
        const wrapper = shallow(<PageComponent />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});

describe("Component Page - check component level UI elements", () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<PageComponent />);
    });

    it("renders the dumb component", () => {
        expect(wrapper.length).toEqual(1);
    });

    it("contains main wrapper block", () => {
        expect(wrapper.find(".js-hook__main-layout").length).toBe(1);
    });
});