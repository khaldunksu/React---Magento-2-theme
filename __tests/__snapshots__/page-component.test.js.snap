// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Page matches snapshot renders page layout correctly 1`] = `
<div>
  <div
    className="container js-hook__main-layout"
  />
</div>
`;
